gui/account/forms.py

```diff
@@ -320,6 +325,15 @@ def clean_bsdusr_home(self):
 
             if home.startswith('/mnt/'):
                 bsdusr_username = self.cleaned_data.get('bsdusr_username', '')
+                volumes = ['/mnt/{}'.format(volume.vol_name) for volume in Volume.objects.all()]
+
+                if len(zfs.list_datasets(path=user_home)) > 0 and \
+                        (len(zfs.list_datasets(path=home)) > 0 and home.startswith(user_home)):
+                    raise forms.ValidationError(_("You cannot change current home directory to "
+                                                  "the dataset inside Your home dataset."))
```

Comment:
Maybe "A dataset inside the home dataset cannot be used as a home directory."

This is still not the greatest... Ideally, we'd suggest the right way to do this, but I don't know what that would be other than just "don't do that".

Reply:
Alternate: "User home directories cannot be set to a a dataset inside the home dataset."

"home dataset" is vague here.  The user's home dataset?

Reply:
I reffered "home dataset" to current home directory.